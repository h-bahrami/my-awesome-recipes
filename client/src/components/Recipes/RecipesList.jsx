/* eslint-disable no-script-url */
import React from 'react';
import {
  Tag, Divider, Table, Spin, Drawer, Button, Icon, Popconfirm,
} from 'antd';
import PropTypes from 'prop-types';
import Recipe from './Recipe';

import {RecipeType} from './Recipe';


export default class RecipesList extends React.Component {
  componentDidMount() {
    const { fetchRecipes } = this.props;
    fetchRecipes();
  }

  getColumns = () => {
    const { selectRecipe, deleteRecipe } = this.props;
    const columns = [
      {
        title: 'Name',
        dataIndex: 'name',
        key: 'name',
      },
      {
        title: 'Vegetarian Food',
        dataIndex: 'vegetarian',
        key: 'vegetarian',
        render: (text, record) => ( record.vegetarian ? 'Yes': 'No')
      },    
      {
        title: '',
        key: 'action',
        render: (text, record) => (
          <span>
            <Button type="primary" shape='round' icon='edit' onClick={() => selectRecipe(record.id)}></Button>
            <Divider type="vertical" />

            <Popconfirm placement="topLeft" title='Really ?' onConfirm={() => deleteRecipe(record.id)} okText="Yes" cancelText="No">
              <Button type="danger" shape='round' icon='delete'></Button>
            </Popconfirm>
          </span>
        ),
      },
    ];
    return columns;
  }

  onClose = () => {
    console.log('onclose')
    const { selectRecipe } = this.props;
    selectRecipe('');
  };

  render() {
    const { loading, items, selectedRecipe, newRecipe, updateRecipe } = this.props;   

    // console.log(loading, items, selectedRecipe);

    return (
      <>
        {/* <Spin spinning={loading} size="large" /> */}
        <Icon type="plus" onClick={(e) => newRecipe()} />
        <Table columns={this.getColumns()} dataSource={items} loading={loading} />

        <Drawer
          title={!selectedRecipe || selectedRecipe.id === '' ? 'Create new Recipe' : `Modifying ${selectedRecipe.name}` } 
          width={520}
          closable={false}
          onClose={this.onClose}
          visible={selectedRecipe} 
          destroyOnClose={true}
        >
          <Recipe
            recipe={selectedRecipe}
            loading={loading}
            onCancel={this.onClose}
            onChange={(x) => updateRecipe(x)}
          />
         
        </Drawer>

      </>
    );
  }
}

RecipesList.propTypes = {
  items: PropTypes.arrayOf(RecipeType),
  selectedRecipe: RecipeType,
  loading: PropTypes.bool,
  fetchRecipes: PropTypes.func.isRequired,
  selectRecipe: PropTypes.func.isRequired,
};

RecipesList.defaultProps = {
  items: [],
  selectedRecipe: null,
  loading: false,
};
