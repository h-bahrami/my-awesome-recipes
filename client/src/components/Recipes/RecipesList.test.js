import React from 'react';
import { shallow } from 'enzyme';
import RecipesList from './RecipesList';
import renderer from 'react-test-renderer';


const recipes = [{
    id: "e941dcb0-d7de-11e9-a822-0b6bc7db7f51",
    name: "Pizza",
    recordTime: "",
    image: "",
    vegeterian: false,
    people: 3,
    ingredients: ["abc", "def"],
    instructions: "abcdefghijk"
},
{
    id: "706cc1f3-3c23-4eff-b02d-cf7764596d71",
    name: "Bitterballen",
    recordTime: "",
    image: "",
    vegeterian: false,
    people: 3,
    ingredients: ["abc", "def"],
    instructions: "abcdefghijk"
}]

describe('Testing RecipesList Component', () => {

    it('renders without crashing', () => {

        const selectRecipe = jest.fn();
        const updateRecipe = jest.fn();
        const deleteRecipe = jest.fn();
        const fetchRecipes = jest.fn();
        const newRecipe = jest.fn();

        const props = {
            items: recipes,
            loading: false,
            selectedRecipe: null,
            error: '',
            selectRecipe,
            updateRecipe,
            deleteRecipe,
            fetchRecipes,
            newRecipe
        };

        shallow(<RecipesList {...props} />);
    });

    it('snapshot test', () => {

        const selectRecipe = jest.fn();
        const updateRecipe = jest.fn();
        const deleteRecipe = jest.fn();
        const fetchRecipes = jest.fn();
        const newRecipe = jest.fn();

        const props = {
            items: recipes,
            loading: false,
            selectedRecipe: null,
            error: '',
            selectRecipe,
            updateRecipe,
            deleteRecipe,
            fetchRecipes,
            newRecipe
        };

        const tree = renderer
            .create(<RecipesList {...props} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('test click event', () => {


        const selectRecipe = jest.fn();
        const updateRecipe = jest.fn();
        const deleteRecipe = jest.fn();
        const fetchRecipes = jest.fn();
        const newRecipe = jest.fn();

        const props = {
            items: recipes,
            loading: false,
            selectedRecipe: null,
            error: '',
            selectRecipe,
            updateRecipe,
            deleteRecipe,
            fetchRecipes,
            newRecipe
        };

        shallow((<RecipesList {...props} />));
        expect(fetchRecipes).toHaveBeenCalled();
    });
});



