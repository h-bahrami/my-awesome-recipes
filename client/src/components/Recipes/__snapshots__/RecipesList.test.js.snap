// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing RecipesList Component snapshot test 1`] = `
Array [
  <div
    className="ant-table-wrapper"
  >
    <div
      className="ant-spin-nested-loading"
    >
      <div
        className="ant-spin-container"
      >
        <div
          className="ant-table ant-table-default ant-table-scroll-position-left"
          style={Object {}}
        >
          <div
            className="ant-table-content"
          >
            <div
              className="ant-table-body"
              onScroll={[Function]}
              onWheel={[Function]}
              style={Object {}}
            >
              <table
                className=""
                style={Object {}}
              >
                <colgroup>
                  <col
                    style={
                      Object {
                        "minWidth": undefined,
                        "width": undefined,
                      }
                    }
                  />
                  <col
                    style={
                      Object {
                        "minWidth": undefined,
                        "width": undefined,
                      }
                    }
                  />
                  <col
                    style={
                      Object {
                        "minWidth": undefined,
                        "width": undefined,
                      }
                    }
                  />
                  <col
                    style={
                      Object {
                        "minWidth": undefined,
                        "width": undefined,
                      }
                    }
                  />
                  <col
                    style={
                      Object {
                        "minWidth": undefined,
                        "width": undefined,
                      }
                    }
                  />
                </colgroup>
                <thead
                  className="ant-table-thead"
                >
                  <tr
                    style={
                      Object {
                        "height": null,
                      }
                    }
                  >
                    <th
                      className=""
                    >
                      <span
                        className="ant-table-header-column"
                      >
                        <div>
                          <span
                            className="ant-table-column-title"
                          >
                            
                          </span>
                          <span
                            className="ant-table-column-sorter"
                          />
                        </div>
                      </span>
                    </th>
                    <th
                      className="ant-table-column-has-actions ant-table-column-has-sorters"
                      onClick={[Function]}
                    >
                      <span
                        className="ant-table-header-column"
                      >
                        <div
                          className="ant-table-column-sorters"
                        >
                          <span
                            className="ant-table-column-title"
                          >
                            Name
                          </span>
                          <span
                            className="ant-table-column-sorter"
                          >
                            <div
                              className="ant-table-column-sorter-inner ant-table-column-sorter-inner-full"
                              title="Sort"
                            >
                              <i
                                aria-label="icon: caret-up"
                                className="anticon anticon-caret-up ant-table-column-sorter-up off"
                              >
                                <svg
                                  aria-hidden="true"
                                  className=""
                                  data-icon="caret-up"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="0 0 1024 1024"
                                  width="1em"
                                >
                                  <path
                                    d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                  />
                                </svg>
                              </i>
                              <i
                                aria-label="icon: caret-down"
                                className="anticon anticon-caret-down ant-table-column-sorter-down off"
                              >
                                <svg
                                  aria-hidden="true"
                                  className=""
                                  data-icon="caret-down"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="0 0 1024 1024"
                                  width="1em"
                                >
                                  <path
                                    d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                  />
                                </svg>
                              </i>
                            </div>
                          </span>
                        </div>
                      </span>
                    </th>
                    <th
                      className="ant-table-column-has-actions ant-table-column-has-sorters"
                      onClick={[Function]}
                    >
                      <span
                        className="ant-table-header-column"
                      >
                        <div
                          className="ant-table-column-sorters"
                        >
                          <span
                            className="ant-table-column-title"
                          >
                            Vegetarian Food
                          </span>
                          <span
                            className="ant-table-column-sorter"
                          >
                            <div
                              className="ant-table-column-sorter-inner ant-table-column-sorter-inner-full"
                              title="Sort"
                            >
                              <i
                                aria-label="icon: caret-up"
                                className="anticon anticon-caret-up ant-table-column-sorter-up off"
                              >
                                <svg
                                  aria-hidden="true"
                                  className=""
                                  data-icon="caret-up"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="0 0 1024 1024"
                                  width="1em"
                                >
                                  <path
                                    d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                  />
                                </svg>
                              </i>
                              <i
                                aria-label="icon: caret-down"
                                className="anticon anticon-caret-down ant-table-column-sorter-down off"
                              >
                                <svg
                                  aria-hidden="true"
                                  className=""
                                  data-icon="caret-down"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="0 0 1024 1024"
                                  width="1em"
                                >
                                  <path
                                    d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                  />
                                </svg>
                              </i>
                            </div>
                          </span>
                        </div>
                      </span>
                    </th>
                    <th
                      className="ant-table-column-has-actions ant-table-column-has-sorters"
                      onClick={[Function]}
                    >
                      <span
                        className="ant-table-header-column"
                      >
                        <div
                          className="ant-table-column-sorters"
                        >
                          <span
                            className="ant-table-column-title"
                          >
                            Added at
                          </span>
                          <span
                            className="ant-table-column-sorter"
                          >
                            <div
                              className="ant-table-column-sorter-inner ant-table-column-sorter-inner-full"
                              title="Sort"
                            >
                              <i
                                aria-label="icon: caret-up"
                                className="anticon anticon-caret-up ant-table-column-sorter-up off"
                              >
                                <svg
                                  aria-hidden="true"
                                  className=""
                                  data-icon="caret-up"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="0 0 1024 1024"
                                  width="1em"
                                >
                                  <path
                                    d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                  />
                                </svg>
                              </i>
                              <i
                                aria-label="icon: caret-down"
                                className="anticon anticon-caret-down ant-table-column-sorter-down off"
                              >
                                <svg
                                  aria-hidden="true"
                                  className=""
                                  data-icon="caret-down"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="0 0 1024 1024"
                                  width="1em"
                                >
                                  <path
                                    d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                  />
                                </svg>
                              </i>
                            </div>
                          </span>
                        </div>
                      </span>
                    </th>
                    <th
                      className=""
                    >
                      <span
                        className="ant-table-header-column"
                      >
                        <div>
                          <span
                            className="ant-table-column-title"
                          >
                            
                          </span>
                          <span
                            className="ant-table-column-sorter"
                          />
                        </div>
                      </span>
                    </th>
                  </tr>
                </thead>
                <tbody
                  className="ant-table-tbody"
                >
                  <tr
                    className="ant-table-row ant-table-row-level-0"
                    data-row-key={0}
                    onClick={[Function]}
                    onContextMenu={[Function]}
                    onDoubleClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    style={
                      Object {
                        "height": null,
                      }
                    }
                  >
                    <td
                      className=""
                      onClick={[Function]}
                    >
                      <span
                        className="ant-avatar ant-avatar-circle ant-avatar-image"
                        style={Object {}}
                      >
                        <img
                          onError={[Function]}
                          src="http://localhost:3001/photos/"
                        />
                      </span>
                    </td>
                    <td
                      className="ant-table-column-has-actions ant-table-column-has-sorters"
                      onClick={[Function]}
                    >
                      Pizza
                    </td>
                    <td
                      className="ant-table-column-has-actions ant-table-column-has-sorters"
                      onClick={[Function]}
                    >
                      No
                    </td>
                    <td
                      className="ant-table-column-has-actions ant-table-column-has-sorters"
                      onClick={[Function]}
                    >
                      
                    </td>
                    <td
                      className=""
                      onClick={[Function]}
                    >
                      <span>
                        <button
                          className="ant-btn ant-btn-primary ant-btn-round ant-btn-icon-only"
                          onClick={[Function]}
                          type="button"
                        >
                          <i
                            aria-label="icon: edit"
                            className="anticon anticon-edit"
                          >
                            <svg
                              aria-hidden="true"
                              className=""
                              data-icon="edit"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 0 0 0-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 0 0 9.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                              />
                            </svg>
                          </i>
                        </button>
                        <div
                          className="ant-divider ant-divider-vertical"
                          role="separator"
                        />
                        <button
                          className="ant-btn ant-btn-danger ant-btn-round ant-btn-icon-only"
                          onClick={[Function]}
                          onMouseDown={[Function]}
                          onTouchStart={[Function]}
                          type="button"
                        >
                          <i
                            aria-label="icon: delete"
                            className="anticon anticon-delete"
                          >
                            <svg
                              aria-hidden="true"
                              className=""
                              data-icon="delete"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
                              />
                            </svg>
                          </i>
                        </button>
                      </span>
                    </td>
                  </tr>
                  <tr
                    className="ant-table-row ant-table-row-level-0"
                    data-row-key={1}
                    onClick={[Function]}
                    onContextMenu={[Function]}
                    onDoubleClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    style={
                      Object {
                        "height": null,
                      }
                    }
                  >
                    <td
                      className=""
                      onClick={[Function]}
                    >
                      <span
                        className="ant-avatar ant-avatar-circle ant-avatar-image"
                        style={Object {}}
                      >
                        <img
                          onError={[Function]}
                          src="http://localhost:3001/photos/"
                        />
                      </span>
                    </td>
                    <td
                      className="ant-table-column-has-actions ant-table-column-has-sorters"
                      onClick={[Function]}
                    >
                      Bitterballen
                    </td>
                    <td
                      className="ant-table-column-has-actions ant-table-column-has-sorters"
                      onClick={[Function]}
                    >
                      No
                    </td>
                    <td
                      className="ant-table-column-has-actions ant-table-column-has-sorters"
                      onClick={[Function]}
                    >
                      
                    </td>
                    <td
                      className=""
                      onClick={[Function]}
                    >
                      <span>
                        <button
                          className="ant-btn ant-btn-primary ant-btn-round ant-btn-icon-only"
                          onClick={[Function]}
                          type="button"
                        >
                          <i
                            aria-label="icon: edit"
                            className="anticon anticon-edit"
                          >
                            <svg
                              aria-hidden="true"
                              className=""
                              data-icon="edit"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 0 0 0-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 0 0 9.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                              />
                            </svg>
                          </i>
                        </button>
                        <div
                          className="ant-divider ant-divider-vertical"
                          role="separator"
                        />
                        <button
                          className="ant-btn ant-btn-danger ant-btn-round ant-btn-icon-only"
                          onClick={[Function]}
                          onMouseDown={[Function]}
                          onTouchStart={[Function]}
                          type="button"
                        >
                          <i
                            aria-label="icon: delete"
                            className="anticon anticon-delete"
                          >
                            <svg
                              aria-hidden="true"
                              className=""
                              data-icon="delete"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
                              />
                            </svg>
                          </i>
                        </button>
                      </span>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <ul
          className="ant-pagination ant-table-pagination"
          style={Object {}}
          unselectable="unselectable"
        >
          <li
            aria-disabled={true}
            className="ant-pagination-disabled ant-pagination-prev"
            onClick={[Function]}
            onKeyPress={[Function]}
            tabIndex={null}
            title="Previous Page"
          >
            <a
              className="ant-pagination-item-link"
            >
              <i
                aria-label="icon: left"
                className="anticon anticon-left"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="left"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 0 0 0 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                  />
                </svg>
              </i>
            </a>
          </li>
          <li
            className="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
            onClick={[Function]}
            onKeyPress={[Function]}
            tabIndex="0"
            title={1}
          >
            <a>
              1
            </a>
          </li>
          <li
            aria-disabled={true}
            className="ant-pagination-disabled ant-pagination-next"
            onClick={[Function]}
            onKeyPress={[Function]}
            tabIndex={null}
            title="Next Page"
          >
            <a
              className="ant-pagination-item-link"
            >
              <i
                aria-label="icon: right"
                className="anticon anticon-right"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
                  />
                </svg>
              </i>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>,
  <button
    className="ant-btn ant-btn-primary ant-btn-round ant-btn-lg"
    onClick={[Function]}
    type="button"
  >
    <i
      aria-label="icon: plus"
      className="anticon anticon-plus"
    >
      <svg
        aria-hidden="true"
        className=""
        data-icon="plus"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
        />
        <path
          d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
        />
      </svg>
    </i>
    <span>
      New Recipe!
    </span>
  </button>,
]
`;
